src/mlpack/methods/hmm/hmm.hpp

```diff
@@ -399,6 +399,7 @@ class HMM
 
   //! Tolerance of Baum-Welch algorithm.
   double tolerance;
+  arma::mat logProbs;
```

Comment:
Awesome, this looks better to me---but I'd suggest passing around `logProbs` between the different methods, so we can keep them `const`.  I say that because I feel that any methods that don't modify the HMM's parameters itself should ideally be `const`.  Anyway that's a minor change that shouldn't make any significant difference.

When you tried this, did it accelerate the implementation further?

Reply:
> Awesome, this looks better to me---but I'd suggest passing around `logProbs` between the different methods, so we can keep them `const`. I say that because I feel that any methods that don't modify the HMM's parameters itself should ideally be `const`. Anyway that's a minor change that shouldn't make any significant difference.
> 
> When you tried this, did it accelerate the implementation further?

No the time bumped up agian, I guess The previous version was better.

Reply:
Hmm, interesting, any chance you can share the code for each revision that you tested?  I wouldn't mind playing with it to see if there's anything that I can figure out.  I would have expected the use of `logProbs` to be faster.

Reply:
The same code as in https://github.com/mlpack/mlpack/pull/2031#issuecomment-534700592

Reply:
@rcurtin friendly-ping 

Bumping up the pr , I hope this didn't fell of your list

Reply:
I appreciate the ping, it is still on my list. :)  This one will take a little while for me to dig into so it's hard to find the time to do it.  I will keep trying...